great_expectations/_docs_decorators.py

```diff
@@ -89,10 +90,28 @@ def my_method(some_argument):
 
     This tag is added at import time.
     """
-    public_api_introspector.add(func)
-    existing_docstring = func.__doc__ if func.__doc__ else ""
-    func.__doc__ = WHITELISTED_TAG + existing_docstring
-    return func
+    public_api_introspector.add(func_or_cls)
+    existing_docstring = func_or_cls.__doc__ if func_or_cls.__doc__ else ""
+    func_or_cls.__doc__ = WHITELISTED_TAG + existing_docstring
+
+    _populate_public_api_registry(func_or_cls)
+    return func_or_cls
+
+
+_public_api_registry = defaultdict(set)
```

Comment:
Don't love globals but on import, this gets populated (see `public_api_registry.json` for output)

Reply:
Could actually leverage existing `_PublicApiIntrospector`